nixos/modules/services/monitoring/watchdogd.nix

```
@@ -0,0 +1,121 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.watchdogd;
+
+  mkPluginOpts = plugin: defWarn: defCrit: {
+    enable = mkEnableOption "watchdogd plugin ${plugin}";
+    interval = mkOption {
+      type = types.ints.unsigned;
+      default = 300;
+      description = ''
+        Amount of seconds between every poll.
+      '';
+    };
+    logmark = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to log current stats every poll interval.
+      '';
+    };
+    warning = mkOption {
+      type = types.ints.unsigned;
+      description = ''
+        The high watermark level. Alert sent to log.
+      '';
+    } // optionalAttrs (defWarn != null) {
+      default = defWarn;
+    };
+    critical = mkOption {
+      type = types.ints.unsigned;
```

On second thought, these two cannot have this type. We even set their defaults to be floats. I'm unsure if they will break without the `.0` suffix on ints.

```suggestion
      type = types.numbers.nonnegative;
      description = ''
        The high watermark level. Alert sent to log.
      '';
    } // optionalAttrs (defWarn != null) {
      default = defWarn;
    };
    critical = mkOption {
      type = types.numbers.nonnegative;
```

---

Good call! I completely overlooked the numbers types.
Thanks!